### PR TITLE
Adding MultiLayerSlider component

### PR DESCRIPTION
**Pull request by weskamm**
## FEATURE 
This adds a MultiLayerSlider component
### Description:
see test, example and description for details

### CODE REVIEW

File: src/Slider/MultiLayerSlider/MultiLayerSlider.jsx [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    this.props.layers[0].setOpacity(1);
+  }
+
+  /**
+   * Formats the tip for the slider.
+   * @param  {Number} value the slider value
+   * @return {String}      the formatted tip value
+   */
+  formatTip(value) {
+    let layerIdx = this.getLayerIndexForSliderValue(value);
```
Comment (KaiVolland) [inline src/Slider/MultiLayerSlider/MultiLayerSlider.jsx]: Plz use const where ever you can.

File: src/Slider/MultiLayerSlider/MultiLayerSlider.jsx [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+      let l = this.props.layers[layerIdx];
+      let layername = l.get('name') || l.get('title');
+      tip = layername + ' ' + opacity + '%';
+      // if (this.props.layers[layerIdx + 1]) {
+      //   let layerRight = this.props.layers[layerIdx + 1].get('name');
+      //   tip += ' &#10; ';
+      //   tip += layerRight + ' ' + Math.abs(opacity - 100) + '%';
+      // }
+
+      // tip = opacity + ' | ' + Math.abs(opacity - 100);
```
Comment (KaiVolland) [inline src/Slider/MultiLayerSlider/MultiLayerSlider.jsx]: Plz remove If not used.

File: src/Slider/MultiLayerSlider/MultiLayerSlider.jsx [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+   * @return {String}      the formatted tip value
+   */
+  formatTip(value) {
+    let layerIdx = this.getLayerIndexForSliderValue(value);
+    let tip;
+    if (this.props.layers[layerIdx]) {
+      let opacity = Math.round(this.props.layers[layerIdx].get('opacity') * 100);
+      let l = this.props.layers[layerIdx];
+      let layername = l.get('name') || l.get('title');
+      tip = layername + ' ' + opacity + '%';
```
Comment (KaiVolland) [inline src/Slider/MultiLayerSlider/MultiLayerSlider.jsx]: Plz use templatestring.

File: src/Slider/MultiLayerSlider/MultiLayerSlider.jsx [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+
+  /**
+   * Called when the value of the slider changed.
+   * @param  {Number} value the new value
+   */
+  valueUpdated(value) {
+    let layerIdx = this.getLayerIndexForSliderValue(value);
+    let opacity = this.getOpacityForValue(value);
+    // set all opacities to 0 first
+    this.props.layers.forEach(l => l.setOpacity(0));
```
Comment (KaiVolland) [inline src/Slider/MultiLayerSlider/MultiLayerSlider.jsx]: Plz extract layers to a const for readability.

File: src/Slider/MultiLayerSlider/MultiLayerSlider.jsx [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+ * @extends React.Component
+ */
+class MultiLayerSlider extends React.Component {
+
+  /**
+   * The className added to this component.
+   * @type {String}
+   * @private
+   */
+  className = 'react-geo-multilayerslider'
```
Comment (annarieger) [inline src/Slider/MultiLayerSlider/MultiLayerSlider.jsx]: missing semicolon (just wondering, why it wasn't caught by eslint..)

File: src/Slider/MultiLayerSlider/MultiLayerSlider.jsx [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+    } = this.props;
+
+    const finalClassName = className
+      ? `${className} ${this.className}`
+      : this.className;
+
+    return (
+      <Slider
+        className={finalClassName}
+        marks={this.getMarks()}
```
Comment (annarieger) [inline src/Slider/MultiLayerSlider/MultiLayerSlider.jsx]: shouldn't the `bind` function be used here (like `this.getMarks().bind(this)`)?

Comment (KaiVolland) [reply src/Slider/MultiLayerSlider/MultiLayerSlider.jsx]: Nice catch.

File: src/Slider/MultiLayerSlider/MultiLayerSlider.jsx [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+
+  /**
+   * The className added to this component.
+   * @type {String}
+   * @private
+   */
+  className = 'react-geo-multilayerslider';
+
+  static propTypes = {
+    className: PropTypes.string,
```
Comment (dnlkoch) [inline src/Slider/MultiLayerSlider/MultiLayerSlider.jsx]: Missing JSDoc.

File: src/Slider/MultiLayerSlider/MultiLayerSlider.jsx [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+
+    /**
+     * The layers that should be handled.
+     * @type {Array}
+     */
+    layers: PropTypes.array.isRequired,
+
+    /**
+     * The default value(s).
```
Comment (dnlkoch) [inline src/Slider/MultiLayerSlider/MultiLayerSlider.jsx]: Could you please explain what defaults will be set here?

File: src/Slider/MultiLayerSlider/MultiLayerSlider.jsx [hunk 9]
```diff
@@ -0,0 +1,166 @@
...
+  className = 'react-geo-multilayerslider';
+
+  static propTypes = {
+    className: PropTypes.string,
+
+    /**
+     * The layers that should be handled.
+     * @type {Array}
+     */
+    layers: PropTypes.array.isRequired,
```
Comment (dnlkoch) [inline src/Slider/MultiLayerSlider/MultiLayerSlider.jsx]: Please validate the array type (ol.layer.base).

File: src/Slider/MultiLayerSlider/MultiLayerSlider.jsx (hunk 6 above)
Comment (weskamm) [reply src/Slider/MultiLayerSlider/MultiLayerSlider.jsx]: No bind needed here, this breaks functionality as there is no function expected. Will be fixed in upcoming PR

**Pull request merged**